scripts/globals/combat/magic_hit_rate.lua

```diff
@@ -42,17 +42,20 @@ xi.combat.magicHitRate.calculateActorMagicAccuracy = function(actor, target, spe
         magicAcc            = magicAcc + elementBonus + affinityBonus
     end
 
-    -- Get dStat Magic Accuracy. NOTE: Ninjutsu does not get this bonus/penalty.
-    if skillType ~= xi.skill.NINJUTSU then
-        if statDiff >= 70 then
-            magicAcc = magicAcc + 30
-        elseif statDiff > 30 then
-            magicAcc = magicAcc + 20 + math.floor((statDiff - 30) / 4)
-        elseif statDiff > 10 then
-            magicAcc = magicAcc + 10 + math.floor((statDiff - 10) / 2)
-        else
-            magicAcc = magicAcc + statDiff
-        end
+    -- Get dStat Magic Accuracy.
+    -- dStat is calculated the same for all types of Magic
+    -- https://wiki.ffo.jp/html/3500.html
+    -- https://wiki.ffo.jp/html/19417.html (Difference in INT validation)
+    if statDiff <= -31 then
+        magicAcc = magicAcc + (-20 + ((statDiff + 30) / 4))
+    elseif statDiff <= -11 then
+        magicAcc = magicAcc + (-10 + ((statDiff + 10) / 2))
+    elseif statDiff > -11 and statDiff < 11 then
+        magicAcc = statDiff
+    elseif statDiff >= 31 then
+        magicAcc = magicAcc + (20 + ((statDiff - 30) / 4))
+    elseif statDiff >= 11 then
+        magicAcc = magicAcc + (10 + ((statDiff - 10) / 2))
```

Comment:
Instead of clamping dStat, I would define a new variable (dStatMacc, for example) and clamp that one.

I intend to restructure this function a little in the future, precisely so we can better debug it. A new variable would allow us to print the exact bonus obtained from dStat.

So final result should look something like this...
```lua
    -- Get dStat Magic Accuracy.
    -- dStat is calculated the same for all types of Magic
    -- https://wiki.ffo.jp/html/3500.html
    -- https://wiki.ffo.jp/html/19417.html (Difference in INT validation)
    local dStatMacc = 0
    
    if statDiff <= -31 then
        dStatMacc = -20 + (statDiff + 30) / 4
    elseif statDiff <= -11 then
        dStatMacc = -10 + (statDiff + 10) / 2
    elseif statDiff >= 31 then
        dStatMacc = 20 + (statDiff - 30) / 4
    elseif statDiff >= 11 then
        dStatMacc = 10 + (statDiff - 10) / 2
    end
    
    magicAcc = magicAcc + utils.clampd(StatMacc, -30, 30)
```

Reply:
Ah good point, will get that change in

Reply:
We did something like this for our debug on ASB - should probably move this conversation somewhere else though.
```lua
-- Variables/steps to calculate finalDamage.
    local spellDamage                 = xi.spells.damage.calculateBaseDamage(caster, target, spell, spellId, skillType, statDiff)
    local multipleTargetReduction     = xi.spells.damage.calculateMTDR(caster, target, spell)
    local eleStaffBonus               = xi.spells.damage.calculateEleStaffBonus(caster, spell, spellElement)
    local magianAffinity              = xi.spells.damage.calculateMagianAffinity(caster, spell)
    local resist                      = xi.spells.damage.calculateResist(caster, target,  spell, skillType, spellElement, statDiff, 0, spellElement)
    local magicBurst                  = xi.spells.damage.calculateIfMagicBurst(caster, target,  spell, spellElement)
    local magicBurstBonus             = xi.spells.damage.calculateIfMagicBurstBonus(caster, target, spell, spellId, spellElement)
    local dayAndWeather               = xi.spells.damage.calculateDayAndWeather(caster, target, spell, spellId, spellElement)
    local magicBonusDiff              = xi.spells.damage.calculateMagicBonusDiff(caster, target, spell, spellId, skillType, spellElement)
    local targetMagicDamageAdjustment = xi.spells.damage.calculateTMDA(caster, target, spellDamageType)
    local divineEmblemMultiplier      = xi.spells.damage.calculateDivineEmblemMultiplier(caster, target, spell)
    local ebullienceMultiplier        = xi.spells.damage.calculateEbullienceMultiplier(caster, target, spell)
    local skillTypeMultiplier         = xi.spells.damage.calculateSkillTypeMultiplier(caster, target, spell, skillType)
    local ninSkillBonus               = xi.spells.damage.calculateNinSkillBonus(caster, target, spell, spellId, skillType)
    local ninFutaeBonus               = xi.spells.damage.calculateNinFutaeBonus(caster, target, spell, skillType)
    local undeadDivinePenalty         = xi.spells.damage.calculateUndeadDivinePenalty(caster, target, spell, skillType)
    local nukeAbsorbOrNullify         = xi.spells.damage.calculateNukeAbsorbOrNullify(caster, target, spell, spellElement)

    -- Debug
    xi.msg.debug(caster, "======================")
    xi.msg.debug(caster, "Final Damage Params")
    xi.msg.debug(caster, "======================")
    xi.msg.debugValue(caster, "Base Spell Damage", spellDamage)
    xi.msg.debugValue(caster, "Multi-target Damage Reduction", multipleTargetReduction)
    xi.msg.debugValue(caster, "Element Staff Bonus", eleStaffBonus)
    xi.msg.debugValue(caster, "Magian Affinity", magianAffinity)
    xi.msg.debugValue(caster, "Magic Resist State", resist)
    xi.msg.debugValue(caster, "Resist Tier Dice Roll", utils.ternary(failedDiceRoll, "Failed", "Success"))
    xi.msg.debugValue(caster, "Magic Burst Multiplier", magicBurst)
    xi.msg.debugValue(caster, "Magic Burst Bonus", magicBurstBonus)
    xi.msg.debugValue(caster, "Day & Weather Bonus", dayAndWeather)
    xi.msg.debugValue(caster, "Magic Bonus Difference", magicBonusDiff)
    xi.msg.debugValue(caster, "Target Magic Damage Adjustment", targetMagicDamageAdjustment)
    xi.msg.debugValue(caster, "Divine Emblem Multipleir", divineEmblemMultiplier)
    xi.msg.debugValue(caster, "Ebullience Multiplier", ebullienceMultiplier)
    xi.msg.debugValue(caster, "Skill Type Multiplier", skillTypeMultiplier)
    xi.msg.debugValue(caster, "Ninjutsu Skill Bonus", ninSkillBonus)
    xi.msg.debugValue(caster, "Ninjutsu Futae Bonus", ninFutaeBonus)
    xi.msg.debugValue(caster, "Undead Divine Penalty", undeadDivinePenalty)
    xi.msg.debugValue(caster, "Nuke Absorb / Nullify", nukeAbsorbOrNullify)

Reply:
I had them asregular prints, couse its pretty much the same when testing locally.
Not opposed to this though, but maybe in a separate PR.

Reply:
The main reason we did it this way was for our testers/devs to test on a beta server at the same time without needing access to the code. I am good either way but yes, definitely a separate PR

Reply:
Fixed.